tests/test_stl_binders.py

```diff
@@ -335,3 +335,21 @@ def test_map_view_types():
     assert type(unordered_map_string_double.items()) is items_type
     assert type(map_string_double_const.items()) is items_type
     assert type(unordered_map_string_double_const.items()) is items_type
+
+
+def test_recursive_containers():
+    recursive_vector = m.RecursiveVector()
+    recursive_vector.append(m.RecursiveVector())
+    recursive_vector[0].append(m.RecursiveVector())
+    recursive_vector[0][0].append(m.RecursiveVector())
+    recursive_vector[0][0].append(m.RecursiveVector())
+    # Can't use len() since test_stl_binders.cpp does not include stl.h,
+    # so the necessary conversion is missing
+    assert recursive_vector[0][0].count(m.RecursiveVector()) == 2
+
+    recursive_map = m.RecursiveMap()
+    recursive_map[1] = m.RecursiveMap()
+    recursive_map[1][1] = m.RecursiveMap()
```

Comment:
Test failures later (new compiler with bugs, refactoring with bugs) will be easier to debug if you only go as deep as you need to. I think that's just one level here. The extra level is is also useful as an additional stress test, but I'd definitely not skip the more minimal easier to debug test. I think it's only a couple lines extra (`assert list(recursive_map[1].keys()) == [1, 2]`).

In a similar vein, although slightly on the paranoid side, using values less special than 0, 1, 2 can help a lot in troubleshooting later (e.g. 101, 202, anything that's more unique, easier to pin-point in a pile of debug output), and less likely to appear by accident.